DAQ/src/StrawAndCaloDigisFromFragments_module.cc

```diff
@@ -405,8 +448,12 @@ void
   event.put(std::unique_ptr<EventNumber_t>(new EventNumber_t( eventNumber )));
   
   // Store the straw digis and calo digis in the event
-  event.put(std::move(straw_digis));
-  event.put(std::move(calo_digis));
+  if (parseTRK_){
+    event.put(std::move(straw_digis));
+  }
+  if (parseCAL_){
+    event.put(std::move(calo_digis));
+  }
 
 }  // produce()
 
```

Comment:
This routine would benefit greatly from major refactoring.   I expect that you inherited a lot of this from the previous author.  As written the code says:

 (loop over the combination of all trk and cal fragments){
     if ( track fragment){
           process the track fragment
     } else if ( cal fragment ) {
          process the cal fragment
     }
 } 

There is no common code between the two modes.  I would write one function to process all of the trk fragments an another to process all of the cal fragments.  I would also extract all of the debug output into helper functions so that the main line of the algorithm is more compact.  It's really hard to tell what this code is doing since it's all debug output with random bits of real code mixed in a few lines at a time.

Reply:
Hi Rob,

We use two different module instances in the Online reco: one for taking care of the Tracker fragments and a second one for the calorimeter. In this way, we don't break/change the naming convention in the reconstruction sequences in the Online running from fragments or from Digis.